Src/Newtonsoft.Json/Linq/JArray.cs

```diff
@@ -176,9 +176,12 @@ public new static JArray Parse(string json, JsonLoadSettings settings)
             {
                 JArray a = Load(reader, settings);
 
-                if (reader.Read() && reader.TokenType != JsonToken.Comment)
+                while (reader.Read())
                 {
-                    throw JsonReaderException.Create(reader, "Additional text found in JSON string after parsing content.");
+                    if (reader.TokenType != JsonToken.Comment)
```

Comment:
No. That is specific to comments in arrays. It really just exists as a backwards compatibility toggle for older bad behavior.